osgp/platform/osgp-adapter-domain-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/domain/da/application/services/AdHocManagementService.java

```diff
@@ -100,9 +100,9 @@ public void handleGetDeviceModelResponse(final GetDeviceModelResponseDto getDevi
         this.responseMessageRouter.send(responseMessage, messageType);
     }
 
-    public void handleGetDataResponse(final String payload, final String deviceIdentification) {
-        LOGGER.info("Handle GET_DATA response for device: {} payload: {}", deviceIdentification, payload);
-        // TODO: further implement handling of GET_DATA messages
+    public void handleGetDataResponse(final ResponseMessage response) {
+        LOGGER.info("Forward GET_DATA response {} for device: {}", response, response.getDeviceIdentification());
+        this.responseMessageRouter.send(response, "GET_DATA");
```

Comment:
We currently need a second parameter, because the response message does not contain the message type.
Using another message type than the one in the response sounds as a design or coding flaw. Perhaps we need it for edge cases.